src/utils/outputs.ts

```diff
@@ -242,3 +242,99 @@ export const getSpeakingNames = (
   });
   return result;
 };
+
+export const getSpeakingNamesWithColors = (
+  speakingNames: SpeakingName[],
+  allColors: Record<string, string>,
+  colors: Record<string, HeisslufType[]>,
+  darkMode: boolean,
+  luminanceSteps: number[],
+  speakingNamesDefaultMapping: SpeakingName[]
+): any => {
+  let result: any = {};
+  let state;
+  const isStateName = (name: string) => {
+    return (
+      name.includes("enabled") ||
+      name.includes("hover") ||
+      name.includes("pressed")
+    );
+  };
+
+  const processState = (name: string) => {
```

Comment:
Same as https://github.com/db-ui/theme-builder/pull/421/files#r1505619609

Move this to an extra function.